src/MongoDB.Analyzer/Core/ExpressionProcessor.cs

```diff
@@ -100,23 +154,17 @@ public static (SyntaxNode RewrittenLinqExpression, ConstantsMapper ConstantsMapp
                       })
                       .ToArray();
 
-                    expressionDescendants = expressionNode.DescendantNodes(n => n != rootNode).OfType<IdentifierNameSyntax>();
-                    rootNodeRemapped = SyntaxFactory.IdentifierName(MqlGeneratorSyntaxElements.Linq.QueryableName);
+                    expressionDescendants = expressionNode.DescendantNodes(n => !rootNodes.Contains(n)).OfType<IdentifierNameSyntax>();
+                    processGenerics = false;
```

Comment:
Node need for setting `processGenerics` here?

Reply:
Done